date.h

```diff
@@ -4140,6 +4162,60 @@ format(const CharT* fmt, const sys_time<Duration>& tp)
                           &abbrev, &offset);
 }
 
+
+// ------------- With ostream as destination -------
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       std::basic_string<CharT, Traits> fmt,
+       const local_time<Duration>& tp)
+{
+    return detail::format(os, std::move(fmt), tp);
+}
+
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       std::basic_string<CharT, Traits> fmt, const sys_time<Duration>& tp)
+{
+    const std::string abbrev("UTC");
+    CONSTDATA std::chrono::seconds offset{0};
+    return detail::format(os, std::move(fmt),
+                          local_time<Duration>{tp.time_since_epoch()}, &abbrev, &offset);
+}
+
+// const CharT* formats
+
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       const CharT* fmt, const local_time<Duration>& tp)
+{
+    return detail::format(os, std::basic_string<CharT>(fmt), tp);
+}
+
+template <class CharT, class Traits, class Duration>
+inline
+std::basic_ostream<CharT, Traits>&
+format(std::basic_ostream<CharT, Traits>& os,
+       const CharT* fmt, const sys_time<Duration>& tp)
+{
+    const std::string abbrev("UTC");
+    CONSTDATA std::chrono::seconds offset{0};
+    return detail::format(os, std::basic_string<CharT>(fmt),
+                          local_time<Duration>{tp.time_since_epoch()},
+                          &abbrev, &offset);
+}
+
+
+// --------- End with ostream as destination -------
```

Comment:
You've done a really nice job with this.  But I still have some potentially show-stopper concerns:

1.  I dislike the `format(cout, ...)` syntax as opposed to `cout << format(...)` syntax.  This by itself is not a huge problem as your code could (I think) be dressed up as a manipulator as is done in `<iomanip>`, which would give it the desired syntax.

2.  Assuming we do the manipulator described in (1), the following code still won't work right:
````
auto t = local_days{sep/17/2016} + 12h + 5min;
std::cout << '|' << setw(30) << left << format("%F %T", t) << "|\n";
std::cout << '|' << setw(30) << right << format("%F %T", t) << "|\n";
````
This should output:
````
|2016-09-17 12:05:00           |
|           2016-09-17 12:05:00|
````
And I'm not fond of just saying:  If you need to respect width and alignment, use format syntax #1 and if not use format syntax #2.  I want one syntax that "just works."

`std::complex` has a similar issue, and the committee solved it by specifying that the `complex` streaming operator has to stream into a `ostringstream`, and then it streams the resulting `string`.  I thought about following that pattern exactly, but some clients will actually want the result as a `string`, and so for them it seems silly to convert the internal `ostringstream` to a `string`, stream it out to another `ostream`, and then the client would have to supply an `ostringstream` and recover the `string` out of that.

I _do_ like some of your implementation, and would like to adopt it regardless of this issue of the `format` API.